src/esmockModule.js

```diff
@@ -13,7 +13,13 @@ import {
 const isObj = o => typeof o === 'object' && o
 const isDefaultIn = o => isObj(o) && 'default' in o
 const isDirPathRe = /^\.?\.?([a-zA-Z]:)?(\/|\\)/
+const isMetaResolve = typeof import.meta.resolve === 'function'
 const nextId = ((id = 0) => () => ++id)()
+const rootDirSlashRe = /^\//
+const protocolFileRe = /^file:/
```

Comment:
s.startsWith('file://') is much more clear than regex. and a bit faster.

Reply:
this jsperf test indicates protocolFileRe.test is faster than s.startsWith https://www.measurethat.net/Benchmarks/Show/4797/1/js-regex-vs-startswith-vs-indexof

s.startsWith would be faster if protocolFileRe weren't stored in a variable, however it is stored in a variable

Reply:
also protocolFileRe.test is safer, because the argument passed into test can be null or undefined, whereas s.startsWith causes a runtime error if s is null or undefined.

additionally, regexps and re.test are used conventionally throughout esmock so both forms should be readable

I prefer protocolFileRe.test over s.startsWith

Reply:
jsperf shows 2x difference )
<img width="470" alt="image" src="https://user-images.githubusercontent.com/20106607/196748088-8e62b500-1b40-4c73-abc3-f2078ea66592.png">

BUT, it's a good argument about personal preference & conventionally usage. Thx!

Reply:
I did read the perf result wrong! you are right, startsWith is faster

Reply:
@koshic this is updated to use `s.startsWith('file://')`, closing as resolved